rfcs/0006-gtfs-canonical-routes.md

```diff
@@ -0,0 +1,125 @@
+- Feature Name: gtfs_canonical_routes
+- Start Date: 2021-06-03
+- RFC PR: [mbta/technology-docs#0006](https://github.com/mbta/technology-docs/pull/0006)
+- Asana task: [🚝 Publish canonical trips/patterns for rail routes](https://app.asana.com/0/881264583703207/1200325279789524/f)
+- Status: Proposed
+
+# Summary
+
+This will enable us to publish in our static GTFS feed "canonical" data for our rail routes. We will
+do so via a "canonical" service that's never active, but which has representative trips with stop
+times and shapes. If a canonical trip is similar enough to a real-world trip that are otherwise
+in the GTFS feed, then it will be made the representative trip of an existing route pattern. If no
```

Comment:
From discussion in the main thread re: the nature of representative trips, it's unclear to me if there's still a case for this. @losvedir could you clarify whether you still think it's relevant to add the "fake trip" as the representative trip for existing route patterns that are made canonical? (in other words, are we talking about an as-is Concept 4️⃣, or your proposed modification?)

Reply:
> @losvedir could you clarify whether you still think it's relevant to add the "fake trip" as the representative trip for existing route patterns that are made canonical?

I still do like the idea of more cleanly separating canonical data from real data. I can't shake the idea that what if someone is using a real trip in a canonical sense, but it doesn't run on the weekend, or it picks up an alert (I think CR trips can have alerts, or they are planning to do that anyway), it may come out funny.

Still, it's not necessary given the definition of route patterns as strictly just stop_ids and orders, and it's only a vague sense. I'm fine with either way that @jfabi decides.

Reply:
> I can't shake the idea that what if someone is using a real trip in a canonical sense, but it doesn't run on the weekend, or it picks up an alert (I think CR trips can have alerts, or they are planning to do that anyway), it may come out funny.

I think we have this problem both ways. In the world where we add the fake trip as the representative one, you'd be in trouble if e.g. you treated the times of the stops as "representative", since they're all just one minute apart (where a real trip would have realistic times). Either way, if you look at properties other than the shape and sequence of stop IDs, there is going to be something "un-representative" about the trip. I think this is an inevitable consequence of trying to fit the concept of a "canonical pattern of service" into a data model designed only to convey actual, concrete service — a solid case for inventing a separate data model a la Concept 5️⃣, though I think it's very reasonable not to go in that direction for now.